pulpcore/cli/common/context.py

```diff
@@ -180,7 +180,16 @@ def call(
         except OpenAPIError as e:
             raise PulpException(str(e))
         except HTTPError as e:
-            raise PulpException(str(e.response.text))
+            try:
+                error_message = str(e.response.json())
+            except JSONDecodeError:
+                error_message = (
+                    "Could not get the result. "
+                    "Received an HTTP {status_code} ({reason}) error.".format(
+                        status_code=e.response.status_code, reason=e.response.reason
+                    )
+                )
+            raise PulpException(error_message)
```

Comment:
Why is this useful? Instead of outputting what we get, we now only output if it decodes to valid json? Also we output it as python repr...

Reply:
I ensured that error messages shown to a user have a similar format regardless of a received response. There were error messages in the JSON format (e.g.: "{'detail': 'You do not have permission to perform this action.'}") and there were error messages that contained an HTML code (see https://github.com/pulp/pulp-cli/pull/549#issuecomment-1235659595).

Do you still think this change is useless? I thought this discrepancy between different types of error messages should be resolved.

Reply:
Hmm. I see your complaint. I believe there are several layers to it.
1. Pulp should not answer with html error messages. We requested json. (Did we [the cli] explicitly set that header?)
2. We should be more clever and translate the errors we know (404, 403, 401, ...) into useful `PulpException`s.
3. This is not really related and should go in a separate PR.

For Errors we did not recognize, i think we should still print verbatim what we get.

Reply:
Okay then, I am removing the proposed change.

Still, the format of errors should be addressed in some way. I would go with option 2.